### PR TITLE
(dsl): Support Update request

### LINKED ISSUE #145: Support Update
### Description:

Enables you to script document updates. The script can update, delete, or skip modifying the document. The update API also supports passing a partial document, which is merged into the existing document.

More about Update API can be found here: https://www.elastic.co/guide/en/elasticsearch/reference/7.17/docs-update.html#docs-update-api-request

### PR DESCRIPTION
**Pull request by dbulaja98**
This PR resolves #145

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala [hunk 1]
```diff
@@ -97,6 +98,30 @@ object ElasticRequest {
       size = None
     )
 
+  def update(index: IndexName, id: DocumentId, script: Script): UpdateRequest =
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala:101]: I would move this down and name it `updateByScript`.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala [hunk 2]
```diff
@@ -462,18 +487,68 @@ object ElasticRequest {
...
+    routing: Option[Routing],
+    script: Option[Script],
+    upsert: Option[Document]
+  ) extends UpdateRequest { self =>
+    def docAsUpsert(value: Boolean): UpdateRequest =
+      self.copy(docAsUpsert = Some(value))
 
-case object AlreadyExists extends CreationOutcome
-case object Created       extends CreationOutcome
+    def docAsUpsertFalse: UpdateRequest =
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala:517]: Let's define `docAsUpsertFalse` and `docAsUpsertTrue` in the trait.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala [hunk 3]
```diff
@@ -462,18 +487,68 @@ object ElasticRequest {
...
+
+      scriptToJson merge docToJson merge docAsUpsertJson merge upsertJson
+    }
+
+    def upsert[A: Schema](doc: A): UpdateRequest =
+      self.copy(upsert = Some(Document.from(doc)))
+  }
+
+  private def getActionAndMeta(requestType: String, parameters: List[(String, Any)]): String =
+    parameters.collect { case (name, Some(value)) => s""""$name" : "${value.toString}"""" }
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala:552]: No need for `.toString`.

File: modules/library/src/main/scala/zio/elasticsearch/request/Document.scala [hunk 4]
```diff
@@ -16,13 +16,16 @@
...
 import zio.schema.Schema
 import zio.schema.codec.JsonCodec
 
-private[elasticsearch] final case class Document(json: String)
+private[elasticsearch] final case class Document(json: Json)
 
 private[elasticsearch] object Document {
   def from[A](doc: A)(implicit schema: Schema[A]): Document = Document(
-    JsonCodec.jsonEncoder(schema).encodeJson(doc, indent = None).toString
+    JsonCodec.jsonEncoder(schema).encodeJson(doc, indent = None).toString.fromJson[Json].fold(_ => Obj(), a => a)
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/request/Document.scala:29]: Can you use `identity` instead of `a => a`?

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

File: modules/library/src/it/scala/zio/elasticsearch/HttpExecutorSpec.scala [hunk 5]
```diff
@@ -1240,6 +1245,60 @@ object HttpExecutorSpec extends IntegrationSpec {
...
+                         index,
+                         documentId,
+                         Script("ctx._source.intField += params['factor']").withParams("factor" -> factor)
+                       )
+                     )
+                doc <- Executor.execute(ElasticRequest.getById(index, documentId)).documentAs[TestDocument]
+              } yield assert(doc)(isSome(equalTo(document.copy(intField = intField + factor))))
+            }
+          },
+          test("successfully create document if it does not exists") {
```
Comment (arnoldlacko) [inline modules/library/src/it/scala/zio/elasticsearch/HttpExecutorSpec.scala:1267]: `exists` -> `exist`

File: modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala (hunk 2 above)
Comment (dbulaja98) [reply modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala:517]: It is already defined.

**Pull request merged**